### PR TITLE
Fixes #12254: pinned addressable gem to ~> 2.3.8

### PR DESCRIPTION
**Pull request by dmitri-d**
This is the last version to support 1.8.7.

### CODE REVIEW

**Comment by mmoll** [comment]
It would be nice to this only for 1.8, as 2.x.x should still be compatible to 2.3.8. The major problem is that we will have a hard time once a distro is updating its package of addressable (i.e. Ubuntu 16.04 or FreeBSD might hit this).

**Comment by dmitri-d** [comment]
I'm very slightly concerned that this might cause strange test failures once the versions deviate enough (although I'm hoping that we'll drop 1.8.7 sooner rather than later). Having said that, this affects bundler-based installs only; for package-based installs we rely on bundler_ext that strips versioning information from gem requirements, forcing smart=proxy to use w/e is available on the system.

**Comment by dmitri-d** [comment]
@mmoll: fixed

**Comment by mmoll** [comment]
I do agree it's anything else than ideal, but in such a situation failing tests would be good IMHO so we know that there's a real action point.

File: bundler.d/test.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@
   gem 'ci_reporter', '>= 1.6.3', "< 2.0.0", :require => false
   gem 'rdoc'
   gem 'test-unit' if RUBY_VERSION > "1.8.7"
+  gem 'addressable', '~> 2.3.8' if RUBY_VERSION == '1.8.7' # 2.4.0 drops support for ruby 1.8.7
+  gem 'addressable' if RUBY_VERSION > '1.8.7'
```
Comment (domcleal) [inline bundler.d/test.rb]: Is this needed?  It's brought in by webmock otherwise.

**Comment by dmitri-d** [comment]
@domcleal: removed unnecessary gem specification.

**Comment by domcleal** [comment]
Merged as 2487519, thanks @witlessbird.

**Pull request closed without merging**